radix-engine/src/system/package/package.rs

```diff
@@ -111,6 +113,37 @@ impl Package {
                 default_auth: input.default_package_access_rule,
             }),
         );
+        {
+            let mut package_event_schema = BTreeMap::<
+                String,
+                BTreeMap<String, (LocalTypeIndex, Schema<ScryptoCustomTypeExtension>)>,
+            >::new();
+            for (blueprint_name, event_schemas) in input.event_schema {
+                let blueprint_schema = package_event_schema.entry(blueprint_name).or_default();
+                for (local_type_index, schema) in event_schemas {
+                    let event_name = {
+                        (*schema
+                            .resolve_type_metadata(local_type_index)
+                            .map_or(
+                                Err(RuntimeError::ApplicationError(
+                                    ApplicationError::EventError(EventError::InvalidEventSchema),
+                                )),
+                                Ok,
+                            )?
+                            .type_name)
+                            .to_owned()
+                    };
+                    blueprint_schema.insert(event_name, (local_type_index, schema));
```

Comment:
Missing duplicate check?

Reply:
I'm a bit confused. We scan an array of `(LocalIndexId, ScryptoSchema)` and extract the event names in the schema. Wouldn't the `insert` overwrite existing entry and automatically de-duplicate?

```
if blueprint_event_schemas.len()
                    != blueprint_event_schemas
                        .keys()
                        .collect::<BTreeSet<_>>()
                        .len()
```

Also, isn't the above check always false?

Reply:
> I'm a bit confused. We scan an array of (LocalIndexId, ScryptoSchema) and extract the event names in the schema.

Ooof, yes you're right. Good catch! Will change.

> Also, isn't the following check always false?

Why would it always be false? If there aren't as many events as unique events names then we should fail. Am i missing something here?

Reply:
`blueprint_event_schemas` is a `BTreeMap` and we're comparing the number of entries with the number of keys.